### PR TITLE
Adding unit tests for proxy create method

### PR DESCRIPTION
**Pull request by s-radyuk**
EES-4156
In this PR:
- adding unit tests for proxy Create method
- adding missing tests for client_test.go TestGetLoadTestsByLabel method
- renaming createRequestBody -> createRequestWrapper

### CODE REVIEW

File: pkg/proxy/proxy_test.go [hunk 1]
```diff
@@ -155,3 +165,82 @@ func buildMocFormReq(requestFiles map[string]string, distributedPods, ltType str
...
+			"unknown",
+			map[string]string{
+				"testFile": "testdata/valid/loadtest.jmx",
+			},
+			http.StatusBadRequest,
+			"{\"error\":\"invalid LoadTest type\"}\n",
+			"application/json; charset=utf-8",
+		}} {
+		t.Run(tt.name, func(t *testing.T) {
+			var logger = zap.NewNop()
```
Comment (vgarvardt) [inline pkg/proxy/proxy_test.go:219]: if you use this

```suggestion
			var logger = zaptest.NewLogger(t)
```

you will get logger output if test fails

File: pkg/proxy/proxy.go [hunk 2]
```diff
@@ -30,17 +30,21 @@ var (
...
 
+type loadTestSpecCreator func(*http.Request, *zap.Logger) (apisLoadTestV1.LoadTestSpec, error)
+
 // Proxy handler
 type Proxy struct {
-	maxLoadTestsRun int
-	kubeClient      *kube.Client
+	maxLoadTestsRun     int
+	kubeClient          *kube.Client
+	httpToSpecConverter loadTestSpecCreator
```
Comment (aleksandrzhiliaev) [inline pkg/proxy/proxy.go:39]: No need to inject this internal function here

File: pkg/proxy/proxy.go [hunk 3]
```diff
@@ -30,17 +30,21 @@ var (
 	ErrFileToStringEmpty error = errors.New("file is empty")
 )
 
+type loadTestSpecCreator func(*http.Request, *zap.Logger) (apisLoadTestV1.LoadTestSpec, error)
```
Comment (aleksandrzhiliaev) [inline pkg/proxy/proxy.go:33]:
```suggestion
var fromHTTPToSpecCreatorFn = fromHTTPRequestToLoadTestSpec
```
then use it as I showed below

File: pkg/proxy/proxy.go [hunk 4]
```diff
@@ -66,7 +70,7 @@ func (p *Proxy) Create(w http.ResponseWriter, r *http.Request) {
 	defer cancel()
 
 	// Making valid LoadTestSpec based on HTTP request
-	ltSpec, err := fromHTTPRequestToLoadTestSpec(r, logger)
+	ltSpec, err := p.httpToSpecConverter(r, logger)
```
Comment (aleksandrzhiliaev) [inline pkg/proxy/proxy.go:73]:
```suggestion
	ltSpec, err := fromHTTPToSpecCreatorFn(r, logger)
```

File: pkg/proxy/proxy.go (hunk 3 above)
Comment (vgarvardt) [reply pkg/proxy/proxy.go:33]: package-level vars is bad idea in general, explicit deps injection is much more straightforward

Comment (aleksandrzhiliaev) [reply pkg/proxy/proxy.go:33]: It doesn't have any shared state to be bad, just another way of dealing with dependencies.

Comment (vgarvardt) [reply pkg/proxy/proxy.go:33]: Just a legal way of monkey-patching for tests 😜

File: pkg/proxy/proxy.go (hunk 2 above)
Comment (s-radyuk) [reply pkg/proxy/proxy.go:39]: I find the current implementation more transparent. It allows me to mock this dependency in tests in a straightforward way.

**Comment by aleksandrzhiliaev** [comment]
+1

**Pull request merged**